src/hooks/useTransition/index.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  useState,
+  useCallback,
+  useLayoutEffect,
+  useEffect,
+  useRef,
+} from 'react'
+import { usePrevious } from '../usePrevious'
+
+export const EXITED = 'exited'
+export const ENTERING = 'entering'
+export const ENTERED = 'entered'
+export const EXITING = 'exiting'
+
+export enum TransitionState {
+  EXITED = 'exited',
+  ENTERING = 'entering',
+  ENTERED = 'entered',
+  EXITING = 'exiting',
+}
+
+interface useTransitionParameters {
+  in: boolean
+  timeout: number
+  defaultTransitionState?: TransitionState
+  onExited?: Function
+  onEntering?: Function
+  onEntered?: Function
+  onExiting?: Function
+}
+
+export const useTransition = ({
+  in: on,
+  defaultTransitionState = TransitionState.EXITED,
+  timeout,
+  onExited = () => {},
+  onEntering = () => {},
+  onEntered = () => {},
+  onExiting = () => {},
+}: useTransitionParameters): [TransitionState, boolean, boolean] => {
+  if (typeof on !== 'boolean') throwError('in')
+  if (typeof timeout !== 'number') throwError('timeout')
```

Comment:
Nit: you can remove these sinds we already have typescript type checking 🤔

Reply:
I considered that as well (removing this gets us to 100% coverage btw 😛 ) however I don't get typescript warnings for wrong args when using the component in Sirius, so I think this still has value.

Reply:
Actually not, if this produces a runtime error as well it should be enough (because it's what we have now). Testing!

Reply:
Fixed ✅